ResponseTimes/LogParser.fs

```diff
@@ -4,55 +4,55 @@ open System
 open System.IO
 open FSharp.Data
 open FSharp.Text.RegexProvider
+open Nessos.Streams
 
-type Logs = CsvProvider<"logs.csv"> 
-type Details = JsonProvider<"details.json", SampleIsList=true>
+type Logs = CsvProvider<"sample.csv"> 
+type Details = JsonProvider<"details.json", SampleIsList = true>
 type ResponseTimeRegex = Regex< @"(?<=Request Finished in )((?<ResponseTime>\d+)(?= milliseconds))", noMethodPrefix = true >
 
-type LogDetails = {
-    ApplicationName : string;
-    LoggerName : string;
-    RequestedURL : Uri;
-    HttpResponseStatusCode : int;
-    HttpVerb : string;
-    Message : string;
-    TransactionId : Option<Guid>;
-    ResponseTime : Option<int>;
-}
+/// Only create the matcher once rather than for every call
+let private matcher = ResponseTimeRegex().Match
```

Comment:
In F#, functions are values. So you can arbitrarily bind a function to a value - it's essentially the same as saying `let private matcher x = ResponseTimeRegex().Match(x)` - in other words, a form of partial function application.